.github/workflows/flowglad-next-sharded-unit-tests.yml

```diff
@@ -0,0 +1,142 @@
+        run: pnpm vitest run --reporter=blob --outputFile=.vitest-reports/blob-${{ matrix.shardIndex }}.json --shard=${{ matrix.shardIndex }}/${{ matrix.shardTotal }}
+      # Upload test results as artifacts for later merging
+      - name: Upload blob report to GitHub Actions Artifacts
+        if: ${{ !cancelled() }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: blob-report-${{ matrix.shardIndex }}
+          path: platform/flowglad-next/.vitest-reports/blob-${{ matrix.shardIndex }}.json
+          include-hidden-files: true
+          retention-days: 1  # Short retention since we merge immediately
+      # Clean up Docker containers after tests complete
+      - name: Run test teardown
+        if: always()  # Run even if tests fail
+        run: pnpm test:teardown
+  
+  # Merges the reports from the sharded tests into a single report.
+  # This job runs after all shards complete and combines their results
+  merge-reports:
+    if: ${{ !cancelled() }}  # Only run if workflow wasn't cancelled
+    needs: [flowglad-next-sharded-tests]  # Wait for all shards to finish
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: platform/flowglad-next
+    steps:
+      # Setup steps for merge job (similar to shard setup)
+      - name: Checkout repository
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          ref: ${{ github.event.inputs.branch || github.ref }}
+      - name: Setup Node.js
+        uses: actions/setup-node@9ced9a43a244f3ac94f13bfd896db8c8f30da67a # v3
+        with:
+          node-version: '18'
+      - name: Install pnpm
+        uses: pnpm/action-setup@41ff72655975bd51cab0327fa583b6e92b6d3061 # v4.2.0
+        with:
+          version: 8
+      - name: Get pnpm store directory
+        shell: bash
+        run: |
+          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
+      - name: Setup pnpm cache
+        uses: actions/cache@6f8efc29b200d32929f49075959781ed54ec270c # v3
+        with:
+          path: ${{ env.STORE_PATH }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
+      - name: Install dependencies
+        run: pnpm install-packages
+      # Download all shard reports from artifacts
+      - name: Download blob reports from GitHub Actions Artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: platform/flowglad-next/.vitest-reports
+          pattern: blob-report-*  # Download all shard reports
+          merge-multiple: true     # Merge into single directory
+      # Debug step to verify all reports were downloaded
+      - name: Verify blob files are ready for merge
+        run: |
+          echo "Blob files in .vitest-reports:"
+          ls -la .vitest-reports/
+          echo "Count of blob files:"
+          find .vitest-reports -name "blob-*.json" | wc -l
```

Comment:
Is the idea that this step doesn't have any semantic effect besides providing output that humans can read in the job logs? If so, it'd be good to explain that a bit more clearly in the comment.